From 404413b25ece5ed11b11012a8eb7685338171c4a Mon Sep 17 00:00:00 2001
From: Wael Karman <wael.karman@gmail.com>
Date: Tue, 8 Aug 2023 08:15:09 +0200
Subject: [PATCH] add desktop wallpaper

---
 clients/desktop-shell.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/clients/desktop-shell.c b/clients/desktop-shell.c
index fb53069..8d6fae1 100644
--- a/clients/desktop-shell.c
+++ b/clients/desktop-shell.c
@@ -776,7 +776,7 @@ background_draw(struct widget *widget, void *data)
 	cr = widget_cairo_create(background->widget);
 	cairo_set_operator(cr, CAIRO_OPERATOR_SOURCE);
 	if (background->color == 0)
-		cairo_set_source_rgba(cr, 0.0, 0.0, 0.2, 1.0);
+		cairo_set_source_rgba(cr, 0.0, 0.0, 0.0, 1.0);
 	else
 		set_hex_color(cr, background->color);
 	cairo_paint(cr);
@@ -786,9 +786,10 @@ background_draw(struct widget *widget, void *data)
 	if (background->image)
 		image = load_cairo_surface(background->image);
 	else if (background->color == 0) {
-		char *name = file_name_with_datadir("pattern.png");
+		char *name = file_name_with_datadir("background.png");//pattern.png
 
 		image = load_cairo_surface(name);
+		background->type = BACKGROUND_SCALE;
 		free(name);
 	}
 
@@ -1541,7 +1542,7 @@ int main(int argc, char *argv[])
 	config_file = weston_config_get_name_from_env();
 	desktop.config = weston_config_parse(config_file);
 	s = weston_config_get_section(desktop.config, "shell", NULL, NULL);
-	weston_config_section_get_bool(s, "locking", &desktop.locking, true);
+	weston_config_section_get_bool(s, "locking", &desktop.locking, false);
 	parse_panel_position(&desktop, s);
 	parse_clock_format(&desktop, s);
 
@@ -1559,6 +1560,7 @@ int main(int argc, char *argv[])
 
 	/* Create panel and background for outputs processed before the shell
 	 * global interface was processed */
+	desktop.panel_position = WESTON_DESKTOP_SHELL_PANEL_POSITION_BOTTOM;
 	if (desktop.want_panel)
 		weston_desktop_shell_set_panel_position(desktop.shell, desktop.panel_position);
 	wl_list_for_each(output, &desktop.outputs, link)
